riscv/src/compiler.rs

```diff
@@ -123,123 +182,120 @@ machine Shift(latch, function_id) {
     ]
 }
 
-impl Register {
-    pub fn new(value: u8) -> Self {
-        Self { value }
-    }
-
-    pub fn is_zero(&self) -> bool {
-        self.value == 0
-    }
-}
-
-impl asm_utils::ast::Register for Register {}
-
-impl fmt::Display for Register {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "x{}", self.value)
-    }
-}
-
-#[derive(Clone, Copy, Debug)]
-pub enum FunctionKind {
-    HiDataRef,
-    LoDataRef,
-}
+/// Compiles riscv assembly to POWDR assembly. Adds required library routines.
```

Comment:
```suggestion
/// Compiles riscv assembly to powdr-asm statements. Adds required library routines.
```

Reply:
Will do that in a subsequent PR to not disturb the move